File: backend/WMSApi/Models/PurchaseOrder.cs
namespace WMSApi.Models;

public enum PurchaseOrderStatus
{
    Submitted = 0,
    Received = 1,
    Closed = 2,
    Cancelled = -1
}

public class PurchaseOrder
{
    public long Id { get; set; }
    public PurchaseOrderStatus Status { get; set; }
    public required ICollection<PurchaseOrderItem> PurchaseOrderItems { get; set; }
    public required DateTime DateCreated { get; set; }
    public DateTime? DateEstimatedDelivery { get; set; }
    public DateTime? DateReceived { get; set; }
    public DateTime? DateLastModified { get; set; }
    public string? Comments { get; set;}
}

File: frontend/wms-client/src/models/purchase-order.ts
import PurchaseOrderItem from "./purchase-order-item";
import PurchaseOrderStatus from "./purchase-order-status";

interface PurchaseOrder {
    id?: number;
    status?: PurchaseOrderStatus;
    purchaseOrderItems: PurchaseOrderItem[];
    dateCreated?: string;
    dateEstimatedDelivery?: string;
    dateReceived?: string;
    dateLastModified?: string;
    comments: string;
}

export default PurchaseOrder;